### ISSUE TITLE
Lzma Uncompression fails with "Memory usage limit was reached"

### ISSUE THREAD

**Issue by pmqs** [pmqs/Compress-Raw-Lzma#1]
Seeing a number of cpantesters failures where the common denominator is the lzma error message `Memory usage limit was reached`

Below is a typical failure
```
PERL_DL_NONLAZY=1 "/home/sand/src/perl/repoperls/installed-perls/host/k93msid/v5.30.0/ac75/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/000prereq.t ......... ok
t/001main.t ........... ok

#     Failed test (t/010main-unzip.t at line 385)
#          got: ''
#     expected: 'abcd1'

#     Failed test (t/010main-unzip.t at line 411)
#          got: undef
#     expected: '5'

#     Failed test (t/010main-unzip.t at line 412)
#          got: '0'
#     expected: '5'
# Uncompression Error: Memory usage limit was reached
```

**Comment by pmqs** [comment]
The [libarchive ](https://github.com/libarchive) library includes a change that covers the same leak. See https://github.com/libarchive/libarchive/commit/8c0c4fd77d961458e690d51098abe1e0e25c9df1

The change in libarchive involves explicitly freeing the `options `data structure

```
	ret = lzma_raw_decoder(&(state->stream), filters);
#if LZMA_VERSION < 50010000
	free(filters[0].options);
#endif
```

**Comment by pmqs** [comment]
Issue reported upstream to the liblzma library  https://sourceforge.net/p/lzmautils/discussion/708858/thread/53c6321fc4/

**Comment by pmqs** [comment]
Fixed by 05c0eac3ab1edd05e5bc945463af004dcbc88c00 and b020045e17a7ab7338c010f5100ec63e8e8e9040
